src/main/java/com/khmall/domain/category/CategoryService.java

```diff
@@ -0,0 +1,154 @@
+import com.khmall.exception.custom.NotFoundException;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class CategoryService {
+
+  private final CategoryRepository categoryRepository;
+  private final CategoryValidator categoryValidator;
+
+  /**
+   * 카테고리를 생성합니다.
+   *
+   * @param request 카테고리 생성 요청
+   * @return 생성된 카테고리 정보
+   * @throws NotFoundException  상위 카테고리가 존재하지 않을 경우
+   * @throws DuplicateException 이미 존재하는 이름의 카테고리일 경우
+   */
+  @Transactional
+  public CategoryResponse createCategory(CategoryCreateRequest request) {
+    Category parent = null;
+    if (request.parentId() != null) {
+      parent = categoryRepository.findById(request.parentId())
+          .orElseThrow(() -> new NotFoundException(CategoryConstants.PARENT_NOT_FOUND));
+    }
+
+    // 상위 카테고리와 이름으로 카테고리 존재 여부 확인
+    categoryValidator.validateDuplicateName(parent, request.name());
+    // 카테고리 깊이 제한 검증
+    categoryValidator.validateDepth(parent);
+
+    // 카테고리 엔티티 생성
+    Category category = CategoryMapper.toEntity(request, parent);
+
+    // 카테고리 저장
+    Category savedCategory = categoryRepository.save(category);
+
+    // 저장된 카테고리 응답으로 변환
+    return CategoryMapper.toResponse(savedCategory);
+  }
+
+  /**
+   * 카테고리를 수정합니다.
+   *
+   * @param id      수정할 카테고리 ID
+   * @param request 카테고리 수정 요청
+   * @return 수정된 카테고리 정보
+   * @throws NotFoundException   카테고리가 존재하지 않을 경우
+   * @throws BadRequestException 잘못된 요청일 경우 (예: 이름이 비어있거나 길이가 초과)
+   * @throws DuplicateException  이미 존재하는 이름의 카테고리일 경우
+   */
+  @Transactional
+  public CategoryResponse updateCategory(Long id, CategoryUpdateRequest request) {
+    Category category = categoryRepository.findById(id)
+        .orElseThrow(() -> new NotFoundException(CategoryConstants.NOT_FOUND));
+
+    // 변경 전 값 백업
+    String prevName = category.getName();
+    Long prevParentId = category.getParent() == null ? null : category.getParent().getCategoryId();
+    Integer prevSortOrder = category.getSortOrder();
+
+    updateCategoryName(category, request);
+    updateParentCategory(category, request, id);
+    updateSortOrder(category, request);
+
+    logCategoryChange(id, prevName, category.getName(), prevParentId,
+        category.getParent() == null ? null : category.getParent().getCategoryId(),
+        prevSortOrder, category.getSortOrder());
+
+    return CategoryMapper.toResponse(category);
+  }
+
+  private void updateCategoryName(Category category, CategoryUpdateRequest request) {
+    request.name().ifPresent(name -> {
+      validateName(name);
+      // 상위 카테고리를 변경 하지 않을 경우 중복 확인
+      if (!request.parentId().isPresent()) {
+        categoryValidator.validateDuplicateName(category.getParent(), name);
+      }
+      category.setName(name);
+    });
```

Comment:
```suggestion
    if(StringUtils.isEmpty(request.name()) {
        return;
    }
    validateName(name);
    // 상위 카테고리를 변경 하지 않을 경우 중복 확인
    if (!request.parentId().isPresent()) {
      categoryValidator.validateDuplicateName(category.getParent(), name);
    }
    category.setName(name);
```
요 방식이 가독성이 좀더 높을 것 같아요.